ui/apps/platform/src/Containers/Policies/Wizard/Step2/PolicyBehaviorForm.tsx

```diff
@@ -127,6 +170,22 @@ function PolicyBehaviorForm() {
             spaceItems={{ default: 'spaceItemsNone' }}
             flexWrap={{ default: 'nowrap' }}
         >
+            <ConfirmationModal
+                ariaLabel="Reset policy criteria"
+                confirmText="Reset policy criteria"
+                isOpen={!!lifeCycleChange}
```

Comment:
Related to the cruel master, can you replace `isOpen={!!lifeCycleChange}` prop with conditional rendering, so TypeScript knows that `lifeCycleChange` is not null? And then remove `| undefined` above?

```js
            {lifeCycleChange && <ConfirmationModal
                ariaLabel="Reset policy criteria"
                confirmText="Reset policy criteria"
                isOpen // I think you need this
                onConfirm={() =>
                    onConfirmChangeLifecycle(
                        lifeCycleChange.lifecycleStage,
                        lifeCycleChange.isChecked
                    )
                }```

Reply:
I'm reluctant to make this change, as that ConfirmationModal is our lightweight wrapper around PatternFly's own `<Modal>` component, and their "pattern" is to always have it in the DOM, and use it's `isOpen` prop to let it control what it renders.

https://www.patternfly.org/v4/components/modal/#modal

Reply:
Can you ask on PatternFly slack channel with a brief explanation of the problem?

That is the same effect as early return in `ModalContent` component:
https://github.com/patternfly/patternfly-react/blob/v5/packages/react-core/src/components/Modal/ModalContent.tsx#L120-L122

However, to be fair, I do not know pro and con of mounting and unmounting a React portal:
https://github.com/patternfly/patternfly-react/blob/v5/packages/react-core/src/components/Modal/Modal.tsx#L243-L262

Reply:
We have been able to solve a few similar problems in TypeScript rewrites of our JavaScript code by lifting up early return based on prop values from within component to conditional rendering of element.